File: frontend/src/features/settings/schemas/togglCredential.ts
import { z } from 'zod'

export const togglCredentialSchema = z.object({
  apiKey: z
    .string()
    .min(1, 'APIキーは必須です')
    .min(32, 'Toggl APIキーは32文字以上である必要があります'),
  
  workspaceId: z
    .number()
    .int('ワークスペースIDは整数である必要があります')
    .positive('ワークスペースIDは正の数である必要があります')
    .optional(),
  
  projectIds: z
    .array(z.number().int().positive())
    .optional(),
  
  defaultTags: z
    .array(z.string().min(1))
    .optional(),
  
  timeZone: z
    .string()
    .optional(),
  
  includeWeekends: z
    .boolean()
    .optional(),
})

export type TogglCredentialFormData = z.infer<typeof togglCredentialSchema>

File: backend/src/main/java/com/example/backend/domain/credentials/toggl/TogglCredential.java
package com.example.backend.domain.credentials.toggl;

import lombok.Builder;
import lombok.Getter;

import java.time.LocalDateTime;
import java.util.List;
import java.util.UUID;

@Getter
@Builder(toBuilder = true)
public class TogglCredential {
    
    private final UUID id;
    private final UUID userId;
    private final String apiKey;
    private final Long workspaceId;
    private final List<Integer> projectIds;
    private final List<String> defaultTags;
    private final String timeZone;
    private final boolean includeWeekends;
    private final boolean isActive;
    private final LocalDateTime createdAt;
    private final LocalDateTime updatedAt;
    
    public boolean hasProjects() {
        return projectIds != null && !projectIds.isEmpty();
    }
    
    public boolean hasDefaultTags() {
        return defaultTags != null && !defaultTags.isEmpty();
    }

    public int getProjectCount() {
        return hasProjects() ? projectIds.size() : 0;
    }
    
    public int getTagCount() {
        return hasDefaultTags() ? defaultTags.size() : 0;
    }
}
